hack/release-from/flags.go

```diff
@@ -0,0 +1,163 @@
+
+var gitRemoteFlag = &cli.StringFlag{
+	Name:    "remote",
+	Usage:   "The git remote to push the release to",
+	Value:   "origin",
+	Sources: cli.EnvVars("GIT_REMOTE"),
+}
+
+var baseOperatorFlag = &cli.StringFlag{
+	Name:    "base-version",
+	Aliases: []string{"base"},
+	Usage: "The version of the operator to base this new version from. " +
+		"It is expected in the format vX.Y.Z for releases and " +
+		"for hashrelease, either vX.Y.Z-n-g<git-hash>-<hashrelease-name> (legacy) or " +
+		"vX.Y.Z-<dev-tag-suffix>-n-g<git-hash>-<product-hashrelease-version> (new) " +
+		"where product-hashrelease-version is in the format vA.B.C-<product-dev-tag-suffix>-u-g<product-git-hash>.",
+	Sources:  cli.EnvVars("OPERATOR_BASE_VERSION"),
+	Required: true,
+	Action: func(ctx context.Context, c *cli.Command, value string) error {
+		if !regexp.MustCompile(fmt.Sprintf(baseVersionFormat, c.String(devTagSuffixFlag.Name))).MatchString(value) {
+			return fmt.Errorf("base-version must be in the format vX.Y.Z or vX.Y.Z-<dev-tag-suffix>-n-g<git-hash>-<hashrelease-name> or " +
+				"vX.Y.Z-<dev-tag-suffix>-n-g<git-hash>-<product-hashrelease-version>")
+		}
+		return nil
+	},
+}
+
+var devTagSuffixFlag = &cli.StringFlag{
+	Name:    "dev-tag-suffix",
+	Usage:   "The suffix used to denote development tags",
+	Sources: cli.EnvVars("DEV_TAG_SUFFIX"),
+	Value:   "0-dev",
+}
+
+var versionFlag = &cli.StringFlag{
+	Name:     "version",
+	Usage:    "The version of the operator to release",
+	Sources:  cli.EnvVars("OPERATOR_VERSION", "VERSION"),
+	Required: true,
+	Action: func(ctx context.Context, c *cli.Command, value string) error {
+		if value == c.String("base-version") {
+			return fmt.Errorf("version cannot be the same as base-version")
+		}
+		if regexp.MustCompile(releaseFormat).MatchString(value) {
+			logrus.Warn("You are releasing a new operator version.")
+			return nil
+		}
+		if !regexp.MustCompile(fmt.Sprintf(hashreleaseFormat, c.String(devTagSuffixFlag.Name))).MatchString(value) {
+			if c.Bool(publishFlag.Name) && c.String(registryFlag.Name) == quayRegistry && c.String(imageFlag.Name) == defaultImageName {
+				return fmt.Errorf("cannot use the default registry and image for publishing operator version %q. "+
+					"Either update registry and/or image flag OR specify version in the format ", value)
+			}
+		}
+		return nil
+	},
+}
+
+var exceptCalicoFlag = &cli.StringSliceFlag{
+	Name: "except-calico",
+	Usage: "A list of Calico images and the version to use for them. " +
+		"This should use the format based on the config/calico_versions.yaml file. " +
+		"e.g. --except-calico calico/cni:vX.Y.Z --except-calico csi-node-driver-registrar:vA.B.C-n-g<git-hash>",
+	Sources: cli.EnvVars("OS_IMAGES_VERSIONS"),
+	Action:  validateOverrides,
+}
+
+var exceptEnterpriseFlag = &cli.StringSliceFlag{
+	Name:    "except-calico-enterprise",
+	Aliases: []string{"except-enterprise", "except-calient"},
+	Usage: "A list of Enterprise images and the versions to use for them. " +
```

Comment:
I think this cli tool is not made to handle long usage strings, as a result the help output is hard to read.